### PR TITLE
sysutils/apcupsd: new dashboard widget

### PR DESCRIPTION
**Pull request by xbb**
Updated the widget for the new dashboard.

Looks about the same as before. I've swapped few arguments around, renamed things, and added some documentation.

image

### CODE REVIEW

**Comment by Gibbon99** [comment]
Here's my version !!

https://forum.opnsense.org/index.php?topic=21487.msg206554#msg206554

**Comment by fichtner** [comment]
@xbb @Gibbon99 how should we handle this case? I'd like to merge a new widget but I don't know which. @xbb worked on apcupsd plugin so I would favour his but it would be nice if you two could coordinate instead of me waltzing in.


Cheers,
Franco

File: sysutils/apcupsd/src/opnsense/www/js/widgets/Apcupsd.js [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+
+        if (status.SELFTEST) {
+            const raw_value = status.SELFTEST.value;
+
+            // self test values
+            const value_keys = {
+                OK: 'status_selftest_ok',
+                BT: 'status_selftest_battery',
+                NG: 'status_selftest_overload',
+                NO: 'status_selftest_no_results',
```
Comment (mickgotwings) [inline sysutils/apcupsd/src/opnsense/www/js/widgets/Apcupsd.js:110]: These do not match with translation keys

**Comment by xbb** [comment]
I'm fine with both options but I think this one is more consistent with what the existing users might expect.

If you want and are ok with it, maybe I could implement also the other variant as a configurable option via widget settings.

File: sysutils/apcupsd/src/opnsense/www/js/widgets/Apcupsd.js (hunk 1 above)
Comment (fichtner) [reply sysutils/apcupsd/src/opnsense/www/js/widgets/Apcupsd.js:110]:
```suggestion
                BT: 'status_selftest_bt',
                NG: 'status_selftest_ng',
                NO: 'status_selftest_no',
```

**Comment by fichtner** [comment]
@xbb good idea, maybe it could show the simplified information from #4156 when space is smaller and expand to this on on extended widget width?

File: sysutils/apcupsd/src/opnsense/www/js/widgets/Apcupsd.js (hunk 1 above)
Comment (xbb) [reply sysutils/apcupsd/src/opnsense/www/js/widgets/Apcupsd.js]: thanks, nice catch

**Comment by fichtner** [comment]
Merging this then to squeeze this into 24.7.1. Thanks!

**Comment by xbb** [comment]
> @xbb good idea, maybe it could show the simplified information from #4156 when space is smaller and expand to this on on extended widget width?

Also yes, I haven't checked yet on how to do that. 

@Gibbon99 let me know also what do you think, sorry I don't check the forums too often, maybe open an issue to discuss it?

**Comment by fichtner** [comment]
discussing on #4156 would be fine too, don't want to close it just yet

**Comment by Monviech** [comment]
Here is a good recent example how to hide and show more information on widget resize. Maybe in this case it could be the heigth to collapse or expand it with more information. I don't know, just an idea for later.

https://github.com/opnsense/plugins/blob/94510a72f3953a2fba121976fc332288e2764d96/dns/ddclient/src/opnsense/www/js/widgets/Dyndns.js#L127-L135

File: sysutils/apcupsd/src/opnsense/www/js/widgets/Apcupsd.js [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import BaseTableWidget from "./BaseTableWidget.js";
```
Comment (Monviech) [inline sysutils/apcupsd/src/opnsense/www/js/widgets/Apcupsd.js:27]: @fichtner https://github.com/opnsense/plugins/pull/4157

That should still be changed.

Comment (fichtner) [reply sysutils/apcupsd/src/opnsense/www/js/widgets/Apcupsd.js:27]: good point, will do it now 👍

**Comment by Gibbon99** [comment]
> > @xbb good idea, maybe it could show the simplified information from #4156 when space is smaller and expand to this on on extended widget width?
> 
> Also yes, I haven't checked yet on how to do that.
> 
> @Gibbon99 let me know also what do you think, sorry I don't check the forums too often, maybe open an issue to discuss it?

Happy to work on something.

Would be nice if the original attribution wasn't lost as part of the old php files disappearing.

eg:

 *    Copyright (C) 2021 Dan Lundqvist
 *    Copyright (C) 2021 David Berry
 *    Copyright (C) 2021 Nicola Pellegrini

Perhaps it could be included the package description.

**Pull request merged**